github/messages.go

```diff
@@ -307,3 +320,25 @@ func ParseWebHook(messageType string, payload []byte) (interface{}, error) {
 	}
 	return event.ParsePayload()
 }
+
+// WebhookTypes returns a list of all the known GitHub event type strings
+// supported by go-github.
+func MessageTypes() []string {
+	types := make([]string, 0, len(eventTypeMapping))
+	for t := range eventTypeMapping {
+		types = append(types, t)
+	}
+	return types
+}
+
+// EventForType returns an empty struct matching the specified GitHub event type.
+// If messageType does not match any known event types, it returns nil.
+func EventForType(messageType string) interface{} {
+	prototype := eventTypeMapping[messageType]
+	if prototype != nil {
+		// return a _copy_ of the pointed-to-object.  Unfortunately, for this
+		// we need to use reflection.
+		return reflect.New(reflect.TypeOf(prototype).Elem()).Interface()
+	}
+	return prototype
```

Comment:
Let's please return the happy path at the end, which is idiomatic Go style:

```suggestion
	if prototype == nil {
		return nil
	}
	// return a _copy_ of the pointed-to-object.  Unfortunately, for this
	// we need to use reflection.
	return reflect.New(reflect.TypeOf(prototype).Elem()).Interface()
```

Reply:
Done.  I also referenced this issue so people could find [the comment above](https://github.com/google/go-github/pull/2865#discussion_r1290433599).